vector/src/main/java/im/vector/home/NewHomeFragment.kt

```diff
@@ -0,0 +1,185 @@
+     * Data management
+     * *********************************************************************************************
+     */
+    /**
+     * Init the rooms data
+     */
+    private fun refreshData(result: HomeRoomsViewModel.Result?) {
+        this.result = result
+        val pinMissedNotifications = PreferencesManager.pinMissedNotifications(activity)
+        val pinUnreadMessages = PreferencesManager.pinUnreadMessages(activity)
+        val notificationComparator = RoomUtils.getNotifCountRoomsComparator(mSession, pinMissedNotifications, pinUnreadMessages)
+        dataUpdateListeners.forEachIndexed { index, listener ->
+            when (index) {
+                ROOM_FRAGMENTS.INVITE.ordinal -> {
+                    listener.onUpdate(mActivity.roomInvitations, notificationComparator)
+                }
+                ROOM_FRAGMENTS.FAVORITE.ordinal -> {
+                    listener.onUpdate(result?.favourites, notificationComparator)
+                }
+                ROOM_FRAGMENTS.NORMAL.ordinal -> {
+                    listener.onUpdate(result?.otherRooms?.plus(result.directChats), notificationComparator)
+                }
+                ROOM_FRAGMENTS.LOW_PRIORITY.ordinal -> {
+                    listener.onUpdate(result?.lowPriorities, notificationComparator)
+                }
+            }
+        }
+        mActivity.hideWaitingView()
+    }
+
+    override fun onLongClickRoom(v: View?, room: Room?, position: Int) {
+        // User clicked on the "more actions" area
+        val tags = room!!.accountData.keys
+        val isFavorite = tags != null && tags.contains(RoomTag.ROOM_TAG_FAVOURITE)
+        val isLowPriority = tags != null && tags.contains(RoomTag.ROOM_TAG_LOW_PRIORITY)
+        RoomUtils.displayPopupMenu(activity, mSession, room, v, isFavorite, isLowPriority, this)
+    }
+
+    override fun onSelectRoom(room: Room?, position: Int) {
+        openRoom(room)
+    }
+
+    override fun onRegister(listener: UpDateListener) {
+        dataUpdateListeners.add(listener)
+    }
+
+    override fun onUnregister(listener: UpDateListener) {
+        dataUpdateListeners.remove(listener)
+    }
+
+    inner class HomePagerAdapter(fm: FragmentManager, val titles: Array<String>) : FragmentPagerAdapter(fm) {
+        private val pinMissedNotifications = PreferencesManager.pinMissedNotifications(activity)
+        private val pinUnreadMessages = PreferencesManager.pinUnreadMessages(activity)
+        private val notificationComparator = RoomUtils.getNotifCountRoomsComparator(mSession, pinMissedNotifications, pinUnreadMessages)
+
+        override fun getItem(position: Int): Fragment {
+            val fragment = when (position) {
+                ROOM_FRAGMENTS.INVITE.ordinal -> {
+                    val fragment = InviteRoomFragment()
+                    fragment.onUpdate(mActivity.roomInvitations, notificationComparator)
```

Comment:
Lines 135 to 139 are the same pattern as in the other blocks.  Please centralise this code into a method.

Reply:
done